### PR TITLE
Add feature switching for the xcodebuild Core Data workaround

### PR DESCRIPTION
**Pull request by mfurtak**
Follow-up to #5626

### CODE REVIEW

File: fastlane_core/spec/project_spec.rb [hunk 1]
```diff
@@ -361,6 +361,8 @@ def count_processes(text)
     describe 'xcodebuild_suppress_stderr option' do
       it 'generates an xcodebuild -list command without stderr redirection by default' do
         project = FastlaneCore::Project.new({ project: "./spec/fixtures/projects/Example.xcodeproj" })
+        require 'pry'
+        binding.pry
```
Comment (mfurtak) [inline fastlane_core/spec/project_spec.rb]: Good thinking @houndci-bot 👍

**Comment by KrauseFx** [comment]
Ha @houndci-bot

@mfurtak: Adding the `clean` would be something we could easily add without having a feature flag: there shouldn't be any harm, and it's not a big change. We're using `clean` in _gym_ and _scan_ as far as I know before building already, now we'd also use it fetching the available schemes.

**Comment by KrauseFx** [comment]
Ha @houndci-bot

@mfurtak: Adding the `clean` would be something we could easily add without having a feature flag: there shouldn't be any harm, and it's not a big change. We're using `clean` in _gym_ and _scan_ as far as I know before building already, now we'd also use it fetching the available schemes.

**Comment by mfurtak** [comment]
Another thing to consider is that we don't think it should be necessary in the long term (if/when Apple fixes the bug).

One of the reasons feature switches were designed as they were was so that we could have clear places to look for things which should be later removed.

If you still think it is likely to be low risk and low impact and the second part isn't important, let me know and I can close this. :+1:

**Comment by KrauseFx** [comment]
I'm ok either way 👍  I just fear no one will enable the feature flag. Have you checked if there is a performance difference between with and without `clean`? I assume it's the same speed

**Comment by mfurtak** [comment]
Yeah, it's always a balance of risk between whether people discover the fix, and possibly breaking something or otherwise making it worse for everyone immediately.

I'll shelve this for now.

**Pull request closed without merging**